### ISSUE TITLE
32-bit build test fails when AVX2 is available

### ISSUE THREAD

**Issue by QuLogic** [minio/sha256-simd#48]
When running on a 64-bit host, a 32-bit build fails tests because AVX2 is available and something does not handle this right:
```
$ go version
go version go1.12.2 linux/amd64

$ GOARCH=amd64 go test
PASS
ok  	github.com/minio/sha256-simd	0.002s

$ GOARCH=386 go test
--- FAIL: TestGolden (0.00s)
    sha256_test.go:2227: AVX2: Sum256 function: sha256() = 6a09e667bb67ae853c6ef372a54ff53a510e527f9b05688c1f83d9ab5be0cd19 want e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
FAIL
exit status 1
FAIL	github.com/minio/sha256-simd	0.001s
```
Running on Fedora 30.

**Comment by QuLogic** [comment]
Note, this was with 0.1.0, but #47 also does not work.

**Comment by QuLogic** [comment]
The previous build that passed was ad98e36, so it was an easy bisect to e529fa194128a50ecb64b49880c294ceea7a46c6.

**Comment by harshavardhana** [comment]
We won't support 32 bit platforms @QuLogic

**Comment by QuLogic** [comment]
Well, you have `*_386.*` files and dispatch for it already, but okay.

If you aren't going to support 32-bit platforms, please say so in your README, and that this is not a 100% drop-in replacement either.

**Comment by AudriusButkevicius** [comment]
Is that by choice or because you don't have time to fix it hence declare not supporting 32bit?

**Comment by harshavardhana** [comment]
We don't have any direct requirement, feel free to send a fix if needed.

**Comment by AudriusButkevicius** [comment]
Does that warrant the issue being closed tho? 

The issue makes it visible that this is something you don't fundamentally object just do not have time to implement. Also gives some visibility to potential users (given currently it's not documented).

**Comment by Stebalien** [comment]
This was a test issue, not a bug. Fix in #50.
